lib/text/vtt_text_parser.js

```diff
@@ -384,6 +385,38 @@ shaka.text.VttTextParser = class {
     }
   }
 
+  /**
+   * Convert color end tag to be valid for xml parsing
+   * For example,
+   * input: <c.yellow.bg_blue>Yellow text on blue bg</c>
+   * output: <c.yellow.bg_blue>Yellow text on blue bg</c.yellow.bg_blue>
+   *
+   * @param {string} payload
+   * @return {string} processed payload
+   */
+  static replaceColorPayload(payload) {
+    const names = [];
+    let nameStart = -1;
+    for (let i = 0; i < payload.length; i++) {
+      if (payload[i] === 'c') {
+        if (payload[i - 1] === '<') {
+          nameStart = i;
+        } else if (payload[i - 1] === '/') {
+          const left = payload.substr(0, i);
+          const right = payload.substr(i + 1);
+          payload = left + names[names.length - 1] + right;
```

Comment:
Actually, now that I think about it, this code has kind of a side effect.
For example, take the given payload...
```
<b>Text1 <i>Text2</b></i>
```
Normally, invalid tags like that would cause the text to be put into the cue verbatim, as it fails to parse.
This code assumes that the tags are always matched, though, and would "correct" the order before parsing to be:
```
<b>Text1 <i>Text2</i></b>
```
So this code might sometimes end up obscuring actual errors. It might be a good idea to check if the first tag matches, here, and return the original payload if it does not.